charms/kfp-api/src/charm.py

```diff
@@ -310,8 +310,7 @@ def _upload_files_to_container(self, config_json):
             raise error
         try:
             file_content = json.dumps(config_json)
-            # no need to add `.json` extension to config file, it is detected automatically
-            config = CONFIG_DIR / "config"
+            config = CONFIG_DIR / "config.json"
```

Comment:
will it work the same for Maria?
or we do not need extension there?

Reply:
It will work for both. That's where workload (apiserver) takes its config and then talks to the DB.
Apparently, my implementation was flawed from the beginning. apiserver used default config. And it took 10-15 minutes to timeout.
Our update-status implementation actually detected it.